hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -148,85 +62,214 @@ public function setComponentStatuses($components) {
+      $replaceWith = $tmp;
+
+      $tmp = $replaceOpenCase;
+      $replaceOpenCase = $replaceOpenCaseWith;
+      $replaceOpenCaseWith = $tmp;
     }
-    $existingCaseType = array('Appraisal');
-    $this->manageCaseTypes($existingCaseType);
-    CRM_Core_Invoke::rebuildMenuAndCaches(TRUE);
 
-    //update query to replace Case with Assignment
+    // Replace case activity types
     $optionGroupID = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionGroup', 'activity_type', 'id', 'name');
-    $sql = "UPDATE civicrm_option_value SET label= replace(label,'Case','Assignment') WHERE label like '%Case%' and option_group_id=$optionGroupID and label <> 'Open Case'";
+    $sql = "UPDATE civicrm_option_value SET label= replace(label, {$replace}, {$replaceWith}) WHERE label like '%{$replace}%' and option_group_id={$optionGroupID} and label <> '{$replaceOpenCase}'";
     CRM_Core_DAO::executeQuery($sql);
 
-    $sql = "UPDATE civicrm_option_value SET label= replace(label,'Open Case','Created New Assignment') WHERE label like '%Case%' and option_group_id=$optionGroupID";
+    // replace (open case) activity type  which is a special case and should be replaced differently
+    $sql = "UPDATE civicrm_option_value SET label= replace(label,'{$replaceOpenCase}', '{$replaceOpenCaseWith}') WHERE label = '{$replaceOpenCase}' and option_group_id={$optionGroupID}";
     CRM_Core_DAO::executeQuery($sql);
-    return TRUE;
   }
 
-  /** function to check if newly added managed entity exist
-   *  If exist then create mapping for that managed entity
-   *  @param array $caseTypes values are case type names (e.g. "Appraisal")
+  /**
+   * Create a defined list of relationship types
+   *
    */
-  public function manageCaseTypes($caseTypes) {
-    $caseType = implode("','", $caseTypes);
-    $sql = "SELECT * from civicrm_case_type WHERE name IN ('{$caseType}')";
-    $caseFetch = CRM_Core_DAO::executeQuery($sql);
-    $values = array();
-    while ($caseFetch->fetch()) {
-      $values[] = "('org.civicrm.hrcase', '{$caseFetch->name}', 'CaseType', {$caseFetch->id}, 'never')";
+  public static function createRelationshipTypes() {
+    $toInsert = '';
+    foreach(self::relationshipsTypesList() as $relationship) {
+      $toInsert .= "('{$relationship['name_a_b']}','{$relationship['name_a_b']}','{$relationship['name_b_a']}','{$relationship['name_b_a']}','{$relationship['name_b_a']}','Individual','Individual',NULL,NULL,0,1),";
     }
-    if (!empty($values)) {
-      $manageSql = "INSERT INTO civicrm_managed (module, name, entity_type, entity_id, cleanup) VALUES " . implode(',', $values) ;
-      CRM_Core_DAO::executeQuery($manageSql);
+    $toInsert = rtrim($toInsert, ',');
+    $sql = "INSERT INTO `civicrm_relationship_type`
+            (`name_a_b`, `label_a_b`, `name_b_a`, `label_b_a`, `description`, `contact_type_a`, `contact_type_b`, `contact_sub_type_a`, `contact_sub_type_b`, `is_reserved`, `is_active`)
+            VALUES {$toInsert}";
```

Comment:
why don't you use the values here directly instead of assigning them to a variable first? Also, the code can get more readable if you split the params into multiple lines instead of these long lines you have